### ISSUE TITLE
TSkTextLayout doesn't recognize BidiMode change in runtime

### ISSUE THREAD

**Issue by havrlisan** [skia4delphi/skia4delphi#183]
One of the greatest features your project offers and the FMX lacks is the RTL text display. Currently, the TSkTextLayout respects the Application.BiDiMode only in its constructor, which is tricky since you cannot recognize BiDiMode change in runtime. The TSkLabel component doesn't have that issue since the RightToLeft property doesn't even exist, and the BiDiMode is checked in the `Draw()` method, and in a better way while at it: 
```delphi
if ((Root = nil) and (Application.BiDiMode = TBiDiMode.bdRightToLeft)) or (TFillTextFlag.RightToLeft in FillTextFlags) then
```
where FillTextFlags adds the RightToLeft flag if
```delphi
Root.BiDiMode = bdRightToLeft
```

My proposal is to either entirely replace the RightToLeft property with that same condition (which seems like a better idea) or to check whether RightToLeft is still up to date on every DoDrawLayout call, for example:
```delphi
procedure TSkTextLayout.DoDrawLayout(const ACanvas: ISkCanvas);
var
  I: Integer;
begin
  if RightToLeft <> (Application.BiDiMode = TBiDiMode.bdRightToLeft) then
    RightToLeft := Application.BiDiMode = TBiDiMode.bdRightToLeft;
  ...
end;
```

Thanks for your time!

**Comment by viniciusfbb** [comment]
Hi @havrlisan! Thank you very much for reporting the issue.

I could implement it that way, but the TTextLayout.RightToLeft property would completely lose the meaning of its existence. The root of this problem is the flaw in the use of the TTextLayout by several controls, which simply do not initialize the RightToLeft property of the TTextLayout. Some controls initialize it, some don't. An example is TText, which is used in TLabel. I reported this to Embarcadero, just add the line:

```delphi
TextLayout.RightToLeft := TFillTextFlag.RightToLeft in FillTextFlags;
```

But, the good news is that the TText case (which is used internally by the label and text of most controls), has been fixed by Embarcadero and will be in the next Alexandria update.
